lib/es-models/item.js

```diff
@@ -0,0 +1,313 @@
+    if (callnum) callnum = callnum.replace(/^\|h/, '')
+    return callnum
+  }
+
+  // ToDO: getting callnum from bib
+  shelfMark () {
+    let callnum = this.callNum()
+    // If not found in var field
+    if (callnum) {
+      // Pull callnumber suffix from fieldTag v if present
+      var callnumSuffix = item.fieldTag('v')
+
+      if (callnumSuffix && callnumSuffix.length) {
+        callnum += ' ' + callnumSuffix[0]
+      }
+
+      // Finally store the complete shelfMark data
+      return [ callnum ]
+    }
+  }
+
+  physicalLocation () {
+    return this.callNum()
+  }
+
+  // trim ? coreObjectsMappingByCode
+  status () {
+    // Availability
+    let status = null
+    const item = this.item
+    // Look for a truthy status.code
+    if (item.isNyplRecord() && item.status && item.status.code && item.status.code.trim()) {
+      status = { path: 'status' }
+      if (item.status.code === '-') {
+        // Code '-' matches both Available and Loaned (!!!) so, check duedate
+        status.id = item.status.duedate ? 'co' : 'a'
+      } else {
+        status.code = item.status.code
+      }
+    } else if (item.varField('876', ['j']).length > 0) {
+      // Mainly serving recap, mapp string statuses:
+      status = { path: '876 $j' }
+      let val = item.varField('876', ['j'])[0].toLowerCase()
+      switch (val) {
+        case 'available': status.id = 'a'; break
+        case 'not available': status.id = 'na'; break
+        case 'loaned': status.id = 'r'; break
+      }
+    }
+    if (status) {
+      // We either know the status id or the status code
+      // Find the right entry in our status mapping based on which we identified:
+      const statusMapped = status.id ? statusMapping[status.id]
+        : utils.coreObjectsMappingByCode(statusMapping, status.code)
+      if (statusMapped) {
+        const statusId = statusMapped.id.split(':').pop()
+        return [{ id: `status:${statusId}`, label: statusMapped.label.trim() }]
+      } else console.error('could not find status for:', status)
+    }
+  }
+
+ // does match just mean == ?
+  aeonUrl () {
+    const code = this.aeonSiteCode()
+    return this.item.bibs.find((bib) => {
+      return bib._aeonUrl() == code
+    })
+  }
+
+  _aeonSiteCode () {
+    return (this.item.fieldTag('x') || [])
+      .filter((noteContent) => noteContent.startsWith('AEON eligible'))
+      .map((noteContent) => noteContent.replace(/^AEON eligible /, ''))
+      .shift()
+  }
+
+  idBarcode () {
+    let barcode = null
+    const item = this.item
+    if (item.barcode) barcode = item.barcode
+    else if (item.varField('876', ['p']).length) barcode = item.varField('876', ['p'])[0]
+    if (barcode) return [barcode]
+  }
+
+  _identifier () {
+    const identifiers = []
+    let shelfMark = this.shelfMark()
+    if (shelfMark) {
+      shelfMark = { value: shelfMark.literal, type: 'bf:ShelfMark' }
+      identifiers.push(shelfMark)
```

Comment:
Note that `this.shelfMark()` returns an array, so you'll want to `concat` it's return. See [how this is handled for checkin-card items](https://github.com/NYPL/research-catalog-indexer/pull/31/files#diff-c90a8c89261e56ad07a2d48471e8a00f3372404a03fd79cb18df5bb4c547affcR227-R238)

Reply:
fixed